d3a_interface/validators/pv_validator.py

```diff
@@ -114,3 +115,22 @@ def validate_rate(cls, **kwargs):
             validate_range_limit(GeneralSettings.RATE_CHANGE_PER_UPDATE_LIMIT.min,
                                  kwargs["energy_rate_decrease_per_update"],
                                  GeneralSettings.RATE_CHANGE_PER_UPDATE_LIMIT.max, error_message)
+
+    @classmethod
+    def validate_settings(cls, **kwargs):
+        """Validator to catch mis-configured PV parameters"""
+        # validate location is provided to work with PV orientations
+        if (kwargs.get("geo_tag_location") is None and kwargs.get("tilt") is not None):
+            raise D3ADeviceException({"misconfiguration": [
+                "Geo-coordinate for PV asset must be provided "
+                "to work with PV orientations."]})
+
+        # validate azimuth isn't out of bound
+        if kwargs.get("azimuth") is not None and not 0 <= kwargs["azimuth"] <= 360:
+            raise D3ADeviceException({"misconfiguration": ["Azimuth could only be in between "
+                                                           "0 and 360 degrees."]})
+
+        # validate tilt isn't out of bound
+        if kwargs.get("tilt") is not None and not 0 <= kwargs["tilt"] <= 90:
+            raise D3ADeviceException({"misconfiguration": ["tilt could only be in between "
```

Comment:
Why not sticking to how we are doing it already, by using `Rangelimit` in the `constants_limits` module, e.g. https://github.com/gridsingularity/d3a-interface/blob/f7ac331728a0b650623796c72f40efae69162c27/d3a_interface/constants_limits.py#L127 ?

Reply:
Thanks, forgot about that, will do it now